tools/tf_env_collect.sh

```diff
@@ -52,84 +67,86 @@ platform.python_compiler(),
 platform.python_implementation(),
 ))
 EOF
-${python_bin_path} /tmp/check_python.py 2>&1  >> ${OUTPUT_FILE}
 
-{
-echo
-echo '== check os platform ==============================================='
-} >> ${OUTPUT_FILE}
+  print_header "check os platform"
 
-cat <<EOF > /tmp/check_os.py
+  "${PYTHON_BIN_PATH}" <<EOF
 import platform
 
-print("""os: %s
-os kernel version: %s
-os release version: %s
-os platform: %s
-linux distribution: %s
-linux os distribution: %s
-mac version: %s
-uname: %s
-architecture: %s
-machine: %s
-""" % (
-platform.system(),
-platform.version(),
-platform.release(),
-platform.platform(),
-platform.linux_distribution(),
-platform.dist(),
-platform.mac_ver(),
-platform.uname(),
-platform.architecture(),
-platform.machine(),
-))
+PLATFORM_ENTRIES = [
+    ("os", "system"),
+    ("os kernel version", "version"),
+    ("os release version", "release"),
+    ("os platform", "platform"),
+    ("freedesktop os release", "freedesktop_os_release"),
+    ("mac version", "mac_ver"),
+    ("uname", "uname"),
+    ("architecture", "architecture"),
+    ("machine", "machine"),
+]
+
+for label, function_name in PLATFORM_ENTRIES:
+    if hasattr(platform, function_name):
```

Comment:
Some machines (like Windows) won't have some of these functions (like "freedesktop_os_release" or "mac_ver"), so this just prints the ones that do exist. I added "freedesktop_os_release" because that's kinda what linux_distribution was, and it lets you know what OS is running inside WSL.